pkg/apis/clouddns/v1/zone_genclient.go

```diff
@@ -0,0 +1,87 @@
+package v1
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/anexia-it/go-anxcloud/pkg/api/types"
+)
+
+func (z *Zone) EndpointURL(ctx context.Context) (*url.URL, error) {
+	u, err := url.ParseRequestURI("/api/clouddns/v1/zone.json/")
+	return u, err
+}
+
+func (z *Zone) DecodeAPIResponse(ctx context.Context, data io.Reader) error {
+	// Declare a custom decoder which allows unknown fields - the Zone struct is not modelling all the fields
+	d := json.NewDecoder(data)
+	return d.Decode(z)
+}
+
+func (z *Zone) FilterAPIRequest(ctx context.Context, req *http.Request) (*http.Request, error) {
+	op, err := types.OperationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// The Update endpoint is NOT at ".../zone.json/{zoneName}", but simply ".../zone.json"
+	if op == types.OperationUpdate {
+		// Strip the appended zoneName from the URL
+		req.URL.Path = path.Dir(req.URL.Path)
+	}
+
+	return req, nil
+}
+
+func (z *Zone) FilterAPIRequestBody(ctx context.Context) (interface{}, error) {
+	op, err := types.OperationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// The Create and Update endpoints expect the Zone's name to be in the request body under the key "zoneName"
```

Comment:
the `name` attribute isn't expected by the API for Create and Update and we probably shouldn't send it, right?

Reply:
Yes, we should be strict in what we send